### PR TITLE
ECM-434 Fix Case API update Multiple reference

### PR DESCRIPTION
**Pull request by cindychmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ECM-434

### Change description ###

Extend Fix Case API to update multiple references in single cases

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    private final MultipleCasesReadingService multipleCasesReadingService;
+
+    @Value("${ccd_gateway_base_url}")
+    private String ccdGatewayBaseUrl;
+
+    public void checkUpdateMultipleReference(CCDRequest ccdRequest, String userToken) {
+        CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+        if (caseData.getEcmCaseType().equals("Multiple")
+                && (caseData.getMultipleReference().length() == 16
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java:26]: One thing to note Cindy is that we have 2 different fields to display multiple references
- multipleReference
- multipleReferenceLinkMarkUp

The multiple reference will always be a 7 digit number. It is the 'LinkMarkUp' that is displaying the 16 digit number. You can use [Case 1800629/2021](https://manage-case.demo.platform.hmcts.net/print/jurisdictions/EMPLOYMENT/case-types/Leeds/cases/1627459424482379) as an example. If you scroll down in the print, you can see it's the LinkMarkUp field that is 16 digits and not the multipleReference

Comment (jthmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java:26]: My approach would be that if the user has entered a multiple reference and the type is set to Multiple then just automatically update the markup field. Don't bother checking if it valid or not - just update it every time

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                || caseData.getMultipleReferenceLinkMarkUp() == null)) {
+            String multipleCaseTypeId = ccdRequest.getCaseDetails().getCaseTypeId() + "_Multiple";
+            List<SubmitMultipleEvent> submitMultipleEvents =
+                    retrieveMultiCase(userToken, multipleCaseTypeId, caseData.getMultipleReference());
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java:30]: Based on the code in ecm-common this is going to be wrong - caseData.getMultipleReference() isn't going to be the 16 digit CCD number. You need caseDetails.caseId

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Value("${ccd_gateway_base_url}")
+    private String ccdGatewayBaseUrl;
+
+    public void checkUpdateMultipleReference(CCDRequest ccdRequest, String userToken) {
+        CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+        if (caseData.getEcmCaseType().equals("Multiple")
+                && (caseData.getMultipleReference().length() == 16
+                || caseData.getMultipleReferenceLinkMarkUp() == null)) {
+            String multipleCaseTypeId = ccdRequest.getCaseDetails().getCaseTypeId() + "_Multiple";
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java:28]: Is there definitely no class that gets you the multiple case id for the corresponding single case type id?

**Pull request closed without merging**